### PR TITLE
[WDB-130] Logging

### PR DESCRIPTION
**Pull request by salmonela7**
Hated this task

### CODE REVIEW

File: backend/RestAPI/Middleware/LoggingMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        {
+            if (httpContext == null)
+            {
+                throw new ArgumentNullException(nameof(httpContext));
+            }
+
+            var sw = new Stopwatch();
+            try
+            {
+
```
Comment (AArlauskas) [inline backend/RestAPI/Middleware/LoggingMiddleware.cs]: @tmaciulis22  you know what to say

File: backend/RestAPI/Middleware/LoggingMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+            httpContext.Response.Body.Seek(0, SeekOrigin.Begin);
+
+            var responseLength = httpContext.Response.Body.Length;
+            if (responseLength is < 20480 and > 0)
+            {
+                responseBody = await new StreamReader(httpContext.Response.Body).ReadToEndAsync();
+            }
+            else if (responseLength > 20480)
+            {
+                responseBody = "Request Body exceeded 20480 characters";
```
Comment (AArlauskas) [inline backend/RestAPI/Middleware/LoggingMiddleware.cs]: Could you explain this magical number?

File: backend/RestAPI/Middleware/LoggingMiddleware.cs [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+            {
+                responseBody = "Empty";
+            }
+
+            httpContext.Response.Body.Seek(0, SeekOrigin.Begin);
+
+            await responseBodyStream.CopyToAsync(originalBodyStream);
+
+            var statusCode = httpContext.Response?.StatusCode;
+            var level = statusCode > 499 ? LogEventLevel.Error : LogEventLevel.Information;
```
Comment (AArlauskas) [inline backend/RestAPI/Middleware/LoggingMiddleware.cs:113]: Interesting way of checking the error code 😄

File: backend/RestAPI/Middleware/LoggingMiddleware.cs [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+            httpContext.Request.Body.Position = 0;
+
+            var queryString = httpContext.Request.QueryString.HasValue 
+                ? httpContext.Request.QueryString.Value 
+                : "Empty";
+
+            Logger.Information(RequestMessageTemplate, httpContext.Request.Method, httpContext.Request.Path, queryString, requestBody);
+        }
+
+
```
Comment (AArlauskas) [inline backend/RestAPI/Middleware/LoggingMiddleware.cs:162]: empty line

**Review by AArlauskas — APPROVED** [review]
It does the trick. Nicely done 👍

**Review by tmaciulis22 — APPROVED** [review]
Looks good 👍🏻

**Pull request merged**